### ISSUE TITLE
Autoexplore avoids unidentified /charming when ^Oka

### ISSUE THREAD

**Issue by Syneil** [crawl/crawl#3091]
**PLATFORM and OS or BROWSER:**
Webtiles, CAO
Google Chrome 112.0.5615.49 (Official Build) (x86_64)
MacOS Ventura 13.3.1 (22E261)

**DCSS VERSION AND GAME INFORMATION:**
0.30-a0-985-g0638232e92

**DESCRIPTION:**
I came across an unidentified wand, while worshipping Okawaru. Using 'o' to autoexplore I noticed that I did not visit this wand, despite it being indicated as a pick-up (with a green border) and being the only pick-up location visible. I manually went to visit the wand and found it was a wand of charming - useless for ^Oka. So, it seems like the autoexplore knew what the item was before my character did.

**Comment by pdpol** [comment]
The behavior here is actually that wands are automatically identified when you step on them. What's happening is autoexplore takes you on top of the tile and ids the wand as useless and continues moving.

**Comment by Syneil** [comment]
Example:

Unidentified wand nearby - green outline indicating it's a visit target
<img width="1680" alt="Screenshot 2023-04-19 at 18 34 50" src="https://user-images.githubusercontent.com/7725171/233155363-13ef9093-967f-4c5b-9357-5b5b6f7def91.png">

Auto-explore does not attempt to visit it
<img width="1680" alt="Screenshot 2023-04-19 at 18 34 56" src="https://user-images.githubusercontent.com/7725171/233155378-e23b4b74-4455-44c6-9351-6c8bc11f12d2.png">

Manually visiting the wand (and then stepping off for the sake of the screenshot) reveals it's an unusable wand of charming
<img width="1680" alt="Screenshot 2023-04-19 at 18 35 02" src="https://user-images.githubusercontent.com/7725171/233155383-f1979fbb-7f2a-47f1-94ba-d40d40d378f7.png">

**Comment by pdpol** [comment]
Hm interesting, I'm not able to repro but maybe someone who's more familiar with this can figure it out!

**Comment by projectNoob** [comment]
> **PLATFORM and OS or BROWSER:** Webtiles, CAO Google Chrome 112.0.5615.49 (Official Build) (x86_64) MacOS Ventura 13.3.1 (22E261)
> 
> **DCSS VERSION AND GAME INFORMATION:** 0.30-a0-985-g0638232e92
> 
> **DESCRIPTION:** I came across an unidentified wand, while worshipping Okawaru. Using 'o' to autoexplore I noticed that I did not visit this wand, despite it being indicated as a pick-up (with a green border) and being the only pick-up location visible. I manually went to visit the wand and found it was a wand of charming - useless for ^Oka. So, it seems like the autoexplore knew what the item was before my character did.

Maybe it is because autoexploring sometimes (but I don't know when to) prefer covering tiles than items. Once it went pass a pile of gold for me.

**Comment by Syneil** [comment]
> Maybe it is because autoexploring sometimes (but I don't know when to) prefer covering tiles than items. Once it went pass a pile of gold for me.

Nope, because when there's nothing else to explore it still won't try to visit the wand - despite having the green border on tiles.
